src/LinAlg/hiopMatrixRajaSparseTriplet.cpp

```diff
@@ -1013,6 +1052,171 @@ hiopMatrixRajaSparseTriplet::RowStartsInfo::~RowStartsInfo()
   alloc.deallocate(idx_start_);
 }
 
+/*
+*  extend original Jac to [Jac -I I]
+*/
+void hiopMatrixRajaSparseTriplet::set_Jac_FR(const hiopMatrixSparse& Jac_c,
+                                             const hiopMatrixSparse& Jac_d,
+                                             int* iJacS,
+                                             int* jJacS,
+                                             double* MJacS)
+{
+  const auto& J_c = dynamic_cast<const hiopMatrixRajaSparseTriplet&>(Jac_c);
+  const auto& J_d = dynamic_cast<const hiopMatrixRajaSparseTriplet&>(Jac_d);
+    
+  // shortcut to the original Jac
+  const int *irow_c = J_c.iRow_;
+  const int *jcol_c = J_c.jCol_;
+  const int *irow_d = J_d.iRow_;
+  const int *jcol_d = J_d.jCol_;
+
+  // assuming original Jac is sorted!
+  int nnz_Jac_c = J_c.numberOfNonzeros();
+  int nnz_Jac_d = J_d.numberOfNonzeros();
+  int m_c = J_c.nrows_;
+  int m_d = J_d.nrows_;
+  int n_c = J_c.ncols_;
+  int n_d = J_d.ncols_;
+  assert(n_c == n_d);
+  assert(ncols_ == n_c + 2*m_c + 2*m_d);
+
+  int nnz_Jac_c_new = nnz_Jac_c + 2*m_c;
+  int nnz_Jac_d_new = nnz_Jac_d + 2*m_d;
+
+  assert(nnz_ == nnz_Jac_c_new + nnz_Jac_d_new);
+  
+  if(J_c.row_starts_host == nullptr){
+    J_c.row_starts_host = J_c.allocAndBuildRowStarts();
+  }
+  assert(J_c.row_starts_host);
+  int* Jc_row_st = J_c.row_starts_host->idx_start_;
+
+  if(J_d.row_starts_host == nullptr){
```

Comment:
space before bracket

Reply:
Fixed. I will wait for others' feedback before pushing the commit.